core/findings.go

```diff
@@ -30,15 +30,23 @@ type Finding struct {
 }
 
 // setupUrls will set the urls used to search through either github or gitlab for inclusion in the finding data
-func (f *Finding) setupUrls(scanType string) {
-	switch scanType {
```

Comment:
I understand why the if statement is here but I am concerned about what happens if they are storing a GHE URL in their config file. How does wraith know if they want to scan GHE or just github.com? They could have a config file set up to store everything but only want to run scans one at a time. How does wraith know which it is? There needs to be some kind of switch on the command line to cleanly allow a user to directly say if they want to scan GHE or Github.com. Also, make sure we are able to scan multiple GHE's. I have been in shops with multiple instances of GHE from different acquisitions.

The code itself is really clean and better than some of mine. The way I implemented this in a previous life was to have a separate command for GHE, that way the user either calls `$ wraith scan-github` or `$ wraith scan-github-enterprise`. I am not sure if that is the right way to go as it brings in a fair amount of duplicate code that may not be necessary, but the concern still stands. How does the tool know what scan to run?

Setting this as a flag or picking it up some other way, will allow you to preserve the scan-type string across the board as you could simply set it to *github-enterprise* instead of *github*. Just food for thought, there is no right or wrong way to do this, it is simply my 4 cents. :+1: 

@funkwhatyouheard

Reply:
I gotcha, haven't been making use of the config files so that's a use case I completely overlooked. For a second I thought you were saying you wanted to support scanning multiple github instances at the same time which sounds like a whole lot of added complexity that I'm not sure would be worth the effort... but now I'm with you. :) 

I'll poke around in a minute and see if I can't get a flag added in there to remove ambiguity.

Reply:
@mattyjones let me know what you think of the implementation I went for. To minimize duplication, added a --enterprise-scan flag and added check where base_url's are determined to ensure it's true. Also added a terminating error in session.go's initialization if one of the two (--enterprise-scan and --enterprise-url) are specified but not the other. Could change it to non-terminating if you think it would play nicer with config files, just wanted it to be clear.